### PR TITLE
PacketIO proto API, metadata, and ports

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: dataplane/internal/kernel/genetlink.go [hunk 1]
```diff
@@ -57,17 +58,23 @@ func NewGenetlinkPort(family, group string) (*GenetlinkPort, error) {
 	}, nil
 }
 
+type PacketMetadata struct {
+	SrcIfIndex int
+	DstIfIndex int
+	Context    int
```
Comment (guoshiuan) [inline dataplane/internal/kernel/genetlink.go:64]: Just curious, what is Context? an int?

File: dataplane/internal/kernel/genetlink.go [hunk 2]
```diff
@@ -57,17 +58,23 @@ func NewGenetlinkPort(family, group string) (*GenetlinkPort, error) {
...
+}
+
 // Writes writes a layer2 frame to the port.
-func (p GenetlinkPort) Write(frame []byte) error {
+func (p GenetlinkPort) Write(frame []byte, md *PacketMetadata) (int, error) {
 	// TODO: Support src, dst, and context
```
Comment (guoshiuan) [inline dataplane/internal/kernel/genetlink.go:69]: Remove this comment.

File: proto/forwarding/forwarding_port.proto [hunk 3]
```diff
@@ -222,3 +223,35 @@ message PortStateRequest {
...
+    ContextId context_id = 1;
+    google.rpc.Status status = 2;
+  }
+}
+
+message HostPortControlMessage {
+  PortDesc port = 1;
+  uint64 port_id = 2;
+  PortId dataplane_port = 3; // ID of the dataplane port related to this one.
+  bool create = 4;
```
Comment (guoshiuan) [inline proto/forwarding/forwarding_port.proto:238]: How is create used?

File: dataplane/internal/kernel/genetlink.go (hunk 1 above)
Comment (DanG100) [reply dataplane/internal/kernel/genetlink.go:64]: just an extra value that the SAI client can configure

File: proto/forwarding/forwarding_port.proto (hunk 3 above)
Comment (DanG100) [reply proto/forwarding/forwarding_port.proto:238]: Set to true if an hostif should be created, false for deletion.

**Review by guoshiuan — APPROVED** [review]

**Pull request merged**